src/util/command_line_handling.cpp

```diff
@@ -666,8 +666,10 @@ namespace hpx { namespace util
             }
         }
 
-        if ((vm.count("hpx:connect") || mode_ == hpx::runtime_mode_connect) &&
-            hpx_host == "127.0.0.1")
+        if ((mode_ == hpx::runtime_mode_console || 
+             mode_ == hpx::runtime_mode_connect || 
+             mode_ == hpx::runtime_mode_worker) 
+            && hpx_host == "127.0.0.1")
```

Comment:
I'm not sure it's even necessary to check the mode - if the ip address alone is 127.0.0.1, then it is probably necessary to set the address correctly. Can it do any harm? I guess the test pass/fail will tell us when it's done.